### ISSUE TITLE
Bounty - Add Hyperliquid Spot Connector

### ISSUE THREAD

**Issue by nikspz** [hummingbot/hummingbot#7262]
### Description

Hummingbot have Hyperliquid_perpetual included in the codebase.
This bounty involves the development and integration of a new Hyperliquid Spot exchange connector into Hummingbot. The new connector should allow users to interact with Hyperliquid’s Spot market, enabling trading functionality such as order creation, tracking, cancellation, and balance monitoring.

**Requirements:**

1. **Connector Features:**
   - Implement buy/sell order creation, including market and limit orders.
   - Ability to track and update orders (e.g., check for filled, partially filled, and canceled orders).
   - Proper order cancellation functionality.
   - Retrieve and display available balance for each asset in the bot.
   - Fetch and display market data (order book, price ticker).
   - Handle API rate limits, errors, and exceptions gracefully.

2. **Integration Details:**
   - Follow Hummingbot's existing framework for connector integration.
   - Use Hyperliquid's API documentation to interact with the exchange’s Spot market.
   - Ensure proper authentication mechanisms are included (e.g., API key setup).
   - Make sure the connector is compatible with Hummingbot's strategies like market-making and arbitrage.
   - Write unit and integration tests to validate order functionality (creation, cancellation, updates).
   - Run functional tests to ensure compatibility with Hyperliquid's API and smooth interaction with Hummingbot’s core.

**Deliverables:**
- Fully functioning **Hyperliquid Spot** connector integrated into Hummingbot.
- Unit and integration tests for the connector.
- Documentation detailing the connector's setup, usage

**Bounty Acceptance Criteria:**
- All features and functionality must be working as described.
- Connector must meet Hummingbot’s coding standards and pass code review.
- Tests and documentation must be clear, complete, and easy to follow.

This bounty is perfect for developers with experience in exchange API integrations and an understanding of Hummingbot’s connector architecture.

### Bounty

- Sponsor: Hummingbot Foundation
- Bounty amount: 300K HBOT
- Developer portion: 300K HBOT

**Comment by aybanda** [comment]
@nikspz  Hi there! I'm really interested in taking on this issue I've got some experience with exchange connectors and I'm familiar with Hummingbot's architecture. I'd love to discuss this further and potentially get assigned to the task. Could we chat about the specifics and timeline? Thanks!

**Comment by isreallee82** [comment]
I did love to work on this but if you feel you are up for this @aybanda then great 👍

**Comment by jaykayudo** [comment]
@nikspz  Can i be assigned this bounty?

**Comment by nikspz** [comment]
Assigned to @isreallee82 please be informed:
- As the exclusive assignee for this bounty, we kindly ask that you keep us updated on your progress over the next few days.
- Please take a moment to review and follow the contribution guidelines [here](https://hummingbot.org/developers/contributions/).
- Once you’ve submitted your PR, be sure to go over the checklist provided [here](https://hummingbot.org/developers/contributions/#checklist).
- If you find yourself unable to continue with the bounty, please let us know so we can reassign it as needed.
- Only one dev can be assigned to one bounty
- Please note that a developer must have exchange access to be eligible for bounty assignment.
- The developer has two weeks to complete the bounty or provide some demonstrable progress.

Thank you for your contribution!

**Comment by isreallee82** [comment]
Pr #7282

**Comment by JonaFly** [comment]
![image](https://github.com/user-attachments/assets/1890ec58-1c8c-42af-b091-2160ac294c39)
Someone knows this issues?

**Comment by nikspz** [comment]
> ![image](https://private-user-images.githubusercontent.com/22541181/400610311-1890ec58-1c8c-42af-b091-2160ac294c39.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyMjcyNjAsIm5iZiI6MTczNjIyNjk2MCwicGF0aCI6Ii8yMjU0MTE4MS80MDA2MTAzMTEtMTg5MGVjNTgtMWM4Yy00MmFmLWIwOTEtMjE2MGFjMjk0YzM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA3VDA1MTYwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjMzYyYjNlMjY0ZmU5Mzc0OTQzOWY0NGQ5ZDYzNWRiZmFlOWJiNjBkYmViNzhiMDlmY2RkODRlNzYwZmVlOWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5RSE_F03XXZ0FD4BrFTcFKMX_sidOPl-9CvG63XtUE8) Someone knows this issues?


hi @JonaFly should work on latest version
![image](https://github.com/user-attachments/assets/c1ad558e-0968-47c1-914c-3740c0e98191)

**Comment by nikspz** [comment]
Thanks for your participation, the bounty has been sent to @isreallee82. The fix has been deployed on development branch and hummingbot version 2.2.0

**Comment by mifth** [comment]
Could you please update the main page for the HL Spot? There is Perp only.

**Comment by nikspz** [comment]
> Could you please update the main page for the HL Spot? There is Perp only.

What do you mean by update  the main page for the HL Spot?

**Comment by nikspz** [comment]
Could you send the link or screenshot?

**Comment by mifth** [comment]
Sure.

![Image](https://github.com/user-attachments/assets/e6df6041-44fa-4f03-8714-5a6dd754a0ef)

**Comment by mifth** [comment]
I also found one possible mistake. I guess the HL supports USDC only?
https://hummingbot.org/exchanges/hyperliquid/#spot-candles-feed

![Image](https://github.com/user-attachments/assets/b6fbc692-47c7-47ae-895f-2a1f9caa40d7)

**Comment by nikspz** [comment]
Good catch, you could submit separate Github ticket for it or PR to fix 
https://hummingbot.org/developers/contributions/
